src/compas/geometry/shapes/capsule.py

```diff
@@ -86,7 +86,7 @@ def line(self):
 
     @line.setter
     def line(self, line):
-        self._line = line
+        self._line = Line(*line)
```

Comment:
while fixing this, i noticed that the capsule has a very different logic than for example the box. it doesn't have a local frame and is defined directly through a line defined by two points. perhaps that should become an alternative constructor. the default, like with the box, should accept a frame and the parameters that define the capsule in that frame. ideally this is the default for all shapes: a frame and parameters. alternative constructors allow to use direct geometrical inputs, but would then have to compute a frame from those internally...

@gonzalocasas @beverlylytle @Licini

Reply:
the suggestion makes sense to me

Reply:
will submit a proposal for this via a different PR